api/spec/src/entitlements/entitlements.tsp

```diff
@@ -0,0 +1,319 @@
+  #suppress "@openmeter/api-spec/casing" "Use existing values"
+  UpdatedAt: "updatedAt",
+}
+
+@route("/api/v1/entitlements")
+@tag("Entitlements (Experimental)")
+interface Entitlements {
+  /**
+   * List all entitlements for all the subjects and features. This endpoint is intended for administrative purposes only.
+   * To fetch the entitlements of a specific subject please use the /api/v1/subjects/{subjectKeyOrID}/entitlements endpoint.
+   * If page is provided that takes precedence and the paginated response is returned.
+   */
+  @get
+  @operationId("listEntitlements")
+  @summary("List all entitlements")
+  list(
+    /**
+     * Filtering by multiple features.
+     *
+     * Usage: `?feature=feature-1&feature=feature-2`
+     */
+    @query feature?: string[],
+
+    /**
+     * Filtering by multiple subjects.
+     *
+     * Usage: `?subject=customer-1&subject=customer-2`
+     */
+    @query subject?: string[],
+
+    /**
+     * Filtering by multiple entitlement types.
+     *
+     * Usage: `?entitlementType=metered&entitlementType=boolean`
+     */
+    @query entitlementType?: EntitlementType[],
+
+    ...OpenMeter.QueryPagination,
+    ...OpenMeter.QueryLimitOffset,
+    ...OpenMeter.QueryOrdering<EntitlementOrderBy>,
+  ): Entitlement[] | OpenMeter.PaginatedResponse<Entitlement> | OpenMeter.CommonErrors;
+
+  /**
+   * Get entitlement by id.
+   */
+  @get
+  @operationId("getEntitlementById")
+  get(
+    @path
+    entitlementId: ULID,
+  ): Entitlement | OpenMeter.CommonErrors | OpenMeter.NotFoundError;
+}
+
+/**
+ * Type of the entitlement.
+ */
+@friendlyName("EntitlementType")
+enum EntitlementType {
+  metered: "metered",
+  boolean: "boolean",
+  static: "static",
+}
+
+// NOTE: use OmitProperties<T> for templates once entitlement models are defined
+
+/**
+ * Entitlement templates are used to define the entitlements of a plan.
+ * Features are omitted from the entitlement template, as they are defined in the rate card.
+ */
+@discriminator("type")
+@friendlyName("Entitlement")
+union Entitlement {
+  metered: EntitlementMetered,
+  static: EntitlementStatic,
+  boolean: EntitlementBoolean,
+}
+
+/**
+ * Shared fields for entitlement creation
+ */
+@friendlyName("EntitlementCreateSharedFields")
+model EntitlementCreateSharedFields {
+  /**
+   * The feature the subject is entitled to use.
+   * Either featureKey or featureId is required.
+   */
+  @example("example-feature-key")
+  featureKey?: Key;
+
+  /**
+   * The feature the subject is entitled to use.
+   * Either featureKey or featureId is required.
+   */
+  @example("01ARZ3NDEKTSV4RRFFQ69G5FAV")
+  featureId?: ULID;
+
+  /**
+   * Additional metadata for the feature.
+   */
+  metadata?: Metadata;
+
+  usagePeriod?: RecurringPeriod;
+}
+
+/**
+ * Create inpurs for metered entitlement
+ */
+@friendlyName("EntitlementMeteredCreateInputs")
+model EntitlementMeteredCreateInputs {
+  ...OmitProperties<EntitlementCreateSharedFields, "usagePeriod">;
```

Comment:
If you're looking for `allOf`, you are limited to extend a single model, but it's available:

```suggestion
model EntitlementMeteredCreateInputs extend OmitProperties<EntitlementCreateSharedFields, "usagePeriod"> {
```

Reply:
why do you think that's more desirable? TypeSpec (as languages usually are) is single inheritance. I'm using multiple types at multiple places... from my point of view I don't see why one should be more important than the rest